lib/libc/math/lib_floorl.c

```diff
@@ -57,11 +57,16 @@
 #ifdef CONFIG_HAVE_LONG_DOUBLE
 long double floorl(long double x)
 {
-	modfl(x, &x);
+	if (isnan(x)) {
+		return NAN;
+	}
+
 	if (x < 0.0) {
 		x -= 1.0;
 	}
 
+	modfl(x, &x);
```

Comment:
Also, should void typecast be added to the modf, modff, modfl calls in all the floor API implementations?
As the modf() will return the integer part of X
What is your opinion ?